internal/core/items/text.rs

```diff
@@ -694,10 +694,10 @@ impl TextInput {
         anchor_pos != cursor_pos
     }
 
-    fn selected_text(self: Pin<&Self>) -> String {
+    fn with_selected_text<T>(self: Pin<&Self>, cb: impl FnOnce(&str) -> T) -> T {
         let (anchor, cursor) = self.selection_anchor_and_cursor();
         let text: String = self.text().into();
-        text.split_at(anchor).1.split_at(cursor - anchor).0.into()
+        cb(text.split_at(anchor).1.split_at(cursor - anchor).0)
```

Comment:
```suggestion
        cb(&text[anchor..cursor])
```
Why make it so complicated?

Reply:
There's no reason, I was just focused on the `&str` part of this patch :)